platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -409,21 +414,26 @@ OtaPalStatus_t otaPal_CreateFileForRx( OtaFileContext_t * const C )
 {
     OtaPalStatus_t result = OTA_PAL_COMBINE_ERR( OtaPalUninitialized, 0 );
     char realFilePath[ OTA_FILE_PATH_LENGTH_MAX ];
-
+    OtaPalPathGenStatus_t status = OtaPalFileGenSuccess;
 
     if( C != NULL )
     {
         if( C->pFilePath != NULL )
         {
             if( C->pFilePath[ 0 ] != ( uint8_t ) '/' )
             {
-                getFilePathFromCWD( realFilePath, ( const char * ) C->pFilePath );
+                status = getFilePathFromCWD( realFilePath, ( const char * ) C->pFilePath );
             }
             else
             {
                 ( void ) strncpy( realFilePath, ( const char * ) C->pFilePath, strlen( ( const char * ) C->pFilePath ) + 1U );
             }
 
+            if( status != OtaPalFileGenSuccess )
+            {
+                LogError( ( "Could not generate the absolute path for the file" ) );
+            }
+
             /* POSIX port using standard library */
             /* coverity[misra_c_2012_rule_21_6_violation] */
             C->pFile = fopen( ( const char * ) realFilePath, "w+b" );
```

Comment:
If `status != OtaPalFileGenSuccess` on line 432, you're still opening an invalid filename here.  You should not be continuing here if the filename isn't valid.

Reply:
Refactored in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1487/commits/a3d852db7cfbb536eddf4de696d8ec0f070a7702